src/prefect/environments/storage/github.py

```diff
@@ -0,0 +1,131 @@
+from typing import TYPE_CHECKING, Any, Dict, List
+
+
+from prefect.environments.storage import Storage
+from prefect.utilities.storage import extract_flow_from_file
+
+if TYPE_CHECKING:
+    from prefect.core.flow import Flow
+
+
+class GitHub(Storage):
+    """
+    GitHub storage class. This class represents the Storage interface for Flows stored
+    in `.py` files in a GitHub repository.
+
+    This class represents a mapping of flow name to file paths contained in the git repo,
+    meaning that all flow files should be pushed independently. A typical workflow using
+    this storage type might look like the following:
+
+    Compose flow `.py` file where flow has GitHub storage:
+
+    ```python
+    flow = Flow("my-flow")
+    flow.storage = GitHub(repo="my/repo", path="/flows/flow.py")
+    ```
+
+    Push this `flow.py` file to the `my/repo` repository under `/flows/flow.py`.
+
+    Call `prefect register -f flow.py` to register this flow with GitHub storage.
+
+    Args:
+        - repo (str): the name of a GitHub repository to store this Flow
+        - path (str, optional): a path pointing to a flow file in the repo
+        - **kwargs (Any, optional): any additional `Storage` initialization options
+    """
+
+    def __init__(self, repo: str, path: str = None, **kwargs: Any) -> None:
+        self.flows = dict()  # type: Dict[str, str]
+        self._flows = dict()  # type: Dict[str, "Flow"]
+        self.repo = repo
+        self.path = path
+
+        super().__init__(**kwargs)
+
+    @property
+    def default_labels(self) -> List[str]:
+        return ["github-flow-storage"]
+
+    def get_flow(self, flow_location: str) -> "Flow":
+        """
+        Given a flow_location within this Storage object, returns the underlying Flow (if possible).
+        If the Flow is not found an error will be logged and `None` will be returned.
+
+        Args:
+            - flow_location (str): the location of a flow within this Storage; in this case,
+                a file path on a repository where a Flow file has been committed
+
+        Returns:
+            - Flow: the requested Flow
+
+        Raises:
+            - UnknownObjectException: if the Flow file is unable to be retrieved
+        """
+        from github import UnknownObjectException
+
+        repo = self._github_client.get_repo(self.repo)
+
+        try:
+            contents = repo.get_contents(flow_location)
```

Comment:
Does this method need the leading `/` to work properly?  Every time I see `/flows/my_flow.py` in our examples here, it looks weird to me (vs. `flows/my_flow.py`)

Reply:
It does not! I will update the example to remove it